### PR TITLE
Rename some Github Actions jobs to make them easier to find

### PR DESCRIPTION
**Pull request by faern**
I found the list on the left hand side of https://github.com/mullvad/mullvadvpn-app/actions a bit hard to parse. With this change, all Rust-related CI jobs will start with "Rust", so hopefully they end up next to each other and are easier to spot.

I also removed "CI" at the end of all names. That felt superfluous

Git checklist:

* [ ] Describe the change in **`CHANGELOG.md`** under the `[Unreleased]` header.
* [ ] Check that commits follow the [Mullvad coding guidelines](https://github.com/mullvad/coding-guidelines)

---
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371)

### CODE REVIEW

**Review by dlon — APPROVED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by dlon — APPROVED** [review]
"Daemon and CLI"? :shrug: 

> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371#-:-Mw70Y3I-iJI7Aal5mlo:b-6hx8pm)* status: :shipit: complete! all files reviewed, all discussions resolved

**Comment by faern** [comment]
This PR stalled. Sorry about that :sweat_smile: 

This is take two! With slightly different naming from before. Now I prefix platform specific workflows with the platform, and if specific to a language, with the language.

Trying to have a bit more consistent naming overall, and with the intent of making it easier to find what you are looking for.

I also remove "CI" from all workflow names since it's kind of redundant.

**Review by albin-mullvad — COMMENTED** [review]
> Reviewed 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371)* status: all files reviewed, 1 unresolved discussion (waiting on @faern)

___
*[.github/workflows/android-xml-tidy.yml, line 1 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371#-Mz-PJhu8SBdYk25XmHM:-Mz-PJhu8SBdYk25XmHN:b-frbzlf) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/c6ed699cb1c9583064b742f5ce23e85d6d366a00/.github/workflows/android-xml-tidy.yml#L1)):*
> ```YAML
> name: Android - XML formatting
> ```

Minor nitpick: I suggest changing to something like "Android - XML format check" since "formatting" could mean that changes are done to the XML file. Same applies to the rust workflow.

**Review by faern — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371)* status: all files reviewed, 1 unresolved discussion (waiting on @albin-mullvad)

___
*[.github/workflows/android-xml-tidy.yml, line 1 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371#-Mz-PJhu8SBdYk25XmHM:-Mz-RNDH1-No1pKkY5OU:b-putebl) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/c6ed699cb1c9583064b742f5ce23e85d6d366a00/.github/workflows/android-xml-tidy.yml#L1)):*
<details><summary><i>Previously, albin-mullvad wrote…</i></summary><blockquote>

Minor nitpick: I suggest changing to something like "Android - XML format check" since "formatting" could mean that changes are done to the XML file. Same applies to the rust workflow.
</blockquote></details>

Hmm. Ok. I thought "check" was implicit since the entire CI is read only. It only checks stuff.

**Review by faern — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371)* status: 8 of 10 files reviewed, 1 unresolved discussion (waiting on @albin-mullvad)

___
*[.github/workflows/android-xml-tidy.yml, line 1 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371#-Mz-PJhu8SBdYk25XmHM:-Mz-Ru5ZCyFg_gUXZ6_l:b64xi8i) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/c6ed699cb1c9583064b742f5ce23e85d6d366a00/.github/workflows/android-xml-tidy.yml#L1)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

Hmm. Ok. I thought "check" was implicit since the entire CI is read only. It only checks stuff.
</blockquote></details>

But fixed. I went with `Check formatting` to use the same type of wording as `Audit dependencies`

**Review by albin-mullvad — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371)* status: 8 of 10 files reviewed, all discussions resolved (waiting on @albin-mullvad)

**Review by dlon — APPROVED** [review]
:lgtm:

> Reviewed 7 of 7 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3371#-:-Mz-TbBD5MrmRNk4DnqG:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved
